Python/pytime.c

```diff
@@ -42,6 +42,27 @@ _PyTime_overflow(void)
                     "timestamp too large to convert to C _PyTime_t");
 }
 
+
+#if defined(MS_WINDOWS) || defined(__APPLE__)
+Py_LOCAL_INLINE(_PyTime_t)
+_PyTime_MulDiv(_PyTime_t ticks, _PyTime_t mul, _PyTime_t div)
+{
+    _PyTime_t sec, nsec;
+    /* Compute ticks * mul / div in two parts to prevent integer overflow:
+       compute the number of seconds ("integer part"), and then the remaining
+       number of nanoseconds ("floating part").
+
+       The caller has to check that ticks * mul, with ticks < div, cannot
```

Comment:
We can support even larger `mul` if represent the fraction `mul / div` as `int + mul2 / div`, where `int = round(mul/div)`. But it seems there is no need to do this now.

Reply:
It may be useful tomorrow if we need to support other weird clocks, but I confirm that I don't see the need for that right now.

CPU frequencies seems to be stuck at 4 GHz for physical reasons, and the maximum mul accepted by win_perf_counter() is (2**63-1)//(10**9) which is larger than 9 GHz. Moreover, recent Windows versions now use a fixed frequency of 10^7 (10 MHz): a resolution of 100 ns. QueryPerformanceFrequency() doesn't seem to be related to the CPU frequency anymore.